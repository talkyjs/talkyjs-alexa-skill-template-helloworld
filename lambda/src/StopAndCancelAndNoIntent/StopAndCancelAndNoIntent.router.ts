import { Router } from "@talkyjs/core";

import { StopAndCancelAndNoIntentScript } from './StopAndCancelAndNoIntent.speech'


export const StopAndCancelAndNoIntentRouter: Router = {
    requestType: "IntentRequest",
    intentName: ["AMAZON.StopIntent","AMAZON.CancelIntent","AMAZON.NoIntent"],
    handler: async (handlerInput) => {
        
        
        const script = new StopAndCancelAndNoIntentScript(handlerInput)
        return script
            .createResponseBuilder()
            .getResponse();
        
        
    }
}

export default StopAndCancelAndNoIntentRouter