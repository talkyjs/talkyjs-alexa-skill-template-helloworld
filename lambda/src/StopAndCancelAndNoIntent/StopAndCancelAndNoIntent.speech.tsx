import React from 'react';
import { SpeechScriptJSX } from '@talkyjs/ssml'

export class StopAndCancelAndNoIntentScript extends SpeechScriptJSX {
    speech() {
        return (
            <speak>
                <p>Good-bye!</p>
            </speak>
        )
    }
    
}