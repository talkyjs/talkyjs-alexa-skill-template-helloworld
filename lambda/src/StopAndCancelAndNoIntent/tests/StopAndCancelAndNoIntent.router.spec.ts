import { HandlerInputCreator } from '@ask-utils/test';
import { RequestHandlerFactory } from '@talkyjs/core';
import { RequestHandler } from 'ask-sdk-core';
import { StopAndCancelAndNoIntentRouter } from '../StopAndCancelAndNoIntent.router'

describe('StopAndCancelAndNoIntentRouter', () => {
  let handler: RequestHandler;
  beforeEach(() => {
    handler = RequestHandlerFactory.create(StopAndCancelAndNoIntentRouter);
  });
  describe('canHandle', () => {
    it('should return false when given a not LaunchRequest', async () => {
      const handlerInput = new HandlerInputCreator().createLaunchRequest();
      await expect(handler.canHandle(handlerInput)).resolves.toEqual(false);
    });
    
    it.each(["AMAZON.StopIntent","AMAZON.CancelIntent","AMAZON.NoIntent"])('should match the snapshot of the %p IntentRequest', async (type) => {
      const handlerInput = new HandlerInputCreator().createIntentRequest({
        name: type,
        confirmationStatus: 'NONE'
      });
      await expect(handler.canHandle(handlerInput)).resolves.toMatchSnapshot();
    });
    
  });
  describe('handle', () => {
    it('should match snapshot', async () => {
      const handlerInput = new HandlerInputCreator().createLaunchRequest();
      await expect(handler.handle(handlerInput)).resolves.toMatchSnapshot();
    });
  });
});
