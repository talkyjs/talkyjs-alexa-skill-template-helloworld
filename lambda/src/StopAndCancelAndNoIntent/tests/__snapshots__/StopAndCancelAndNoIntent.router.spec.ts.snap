// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopAndCancelAndNoIntentRouter canHandle should match the snapshot of the "AMAZON.CancelIntent" IntentRequest 1`] = `true`;

exports[`StopAndCancelAndNoIntentRouter canHandle should match the snapshot of the "AMAZON.NoIntent" IntentRequest 1`] = `true`;

exports[`StopAndCancelAndNoIntentRouter canHandle should match the snapshot of the "AMAZON.StopIntent" IntentRequest 1`] = `true`;

exports[`StopAndCancelAndNoIntentRouter handle should match snapshot 1`] = `
Object {
  "outputSpeech": Object {
    "ssml": "<speak><p>Good-bye!</p></speak>",
    "type": "SSML",
  },
}
`;
